### PR TITLE
Add code to build.ps1 to build the installer (not called yet)

### PR DESCRIPTION
**Pull request by tristanlabelle**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -898,6 +932,41 @@ function Build-SourceKitLSP($Arch)
...
 }
 
+function Build-Installer()
+{
+  Build-WiXProject toolchain.wixproj -Properties @{
+    DEVTOOLS_ROOT = "$ToolchainInstallRoot\";
+    TOOLCHAIN_ROOT = "$ToolchainInstallRoot\";
+  }
+
+  msbuild $SourceCache\swift-installer-scripts\platforms\Windows\CustomActions\SwiftInstaller\SwiftInstaller.vcxproj -t:restore
```
Comment (compnerd) [inline build.ps1:942]: This will go away with https://github.com/apple/swift-installer-scripts/pull/176

File: build.ps1 [hunk 2]
```diff
@@ -898,6 +932,41 @@ function Build-SourceKitLSP($Arch)
...
+  Build-WiXProject sdk.wixproj -Properties @{
+    PlatformToolset = "v143";
```
Comment (compnerd) [inline build.ps1:944]: This will go away with https://github.com/apple/swift-installer-scripts/pull/176

File: build.ps1 [hunk 3]
```diff
@@ -898,6 +932,41 @@ function Build-SourceKitLSP($Arch)
...
+
+  Build-WiXProject runtime.wixproj -Properties @{
+    SDK_ROOT = "$SDKInstallRoot\";
+  }
+
+  Build-WiXProject icu.wixproj -Properties @{
+    ProductVersion = "69.1";
+    ProductVersionMajor = "69";
+    ICU_ROOT = "S:\";
+  }
```
Comment (compnerd) [inline build.ps1:958]: This should no longer be needed on 5.8+ as ICU is statically linked into Foundation.

**Review by compnerd — CHANGES_REQUESTED** [review]
Can we split up the renaming and tweaks into a separate PR so that we can get that merged early?

File: build.ps1 (hunk 2 above)
Comment (tristanlabelle) [reply build.ps1:944]: The `PlatformToolset` was for the `SwiftInstaller.vcxproj`?

Comment (tristanlabelle) [reply build.ps1]: Removed

File: build.ps1 (hunk 1 above)
Comment (tristanlabelle) [reply build.ps1]: Removed

File: build.ps1 [hunk 4]
```diff
@@ -226,6 +236,39 @@ function Build-CMakeProject
...
+  param(
+    [Parameter(Position = 0)]
+    [string]$FileName,
+    [hashtable]$Properties = @{}
+  )
+
+  $WiXVersion = "3.11.2"
+  if (-not (Test-Path "$BinaryCache\WiX.$WiXVersion"))
+  {
+    nuget.exe install WiX -Version $WiXVersion -Source https://api.nuget.org/v3/index.json -OutputDirectory $BinaryCache
```
Comment (compnerd) [inline build.ps1:251]: I think that this is fine for checkpointing, but I think that we want to integrate that into the wixproj files themselves, so you can just do a `-t:restore` with the project before we do the build.

File: build.ps1 [hunk 5]
```diff
@@ -226,6 +236,39 @@ function Build-CMakeProject
...
+  TryAdd-KeyValue $Properties OutputPath $BinaryCache\msi\
+  TryAdd-KeyValue $Properties IntermediateOutputPath BinaryCache\$Name\
+
+  $MsbuildArgs = @("$SourceCache\swift-installer-scripts\platforms\Windows\$FileName")
+  $MsbuildArgs += "-noLogo"
+  foreach ($Property in $Properties.GetEnumerator()) {
+    $MsbuildArgs += "-p:$($Property.Key)=$($Property.Value)"
+  }
+
+  & $msbuild @MsbuildArgs
```
Comment (compnerd) [inline build.ps1:268]:
```suggestion
  $MSBuildArgs = @("$SourceCache\swift-installer-scripts\platforms\Windows\$FileName")
  $MSBuildArgs += "-noLogo"
  foreach ($Property in $Properties.GetEnumerator()) {
    $MSBuildArgs += "-p:$($Property.Key)=$($Property.Value)"
  }

  & $msbuild @MSBuildArgs
```

File: build.ps1 [hunk 6]
```diff
@@ -226,6 +236,39 @@ function Build-CMakeProject
...
+
+  $Name = $FileName.Split('.')[0]
+
+  $Properties = $Properties.Clone()
+  TryAdd-KeyValue $Properties RunWixToolsOutOfProc true
+  TryAdd-KeyValue $Properties OutputPath $BinaryCache\msi\
+  TryAdd-KeyValue $Properties IntermediateOutputPath BinaryCache\$Name\
+
+  $MsbuildArgs = @("$SourceCache\swift-installer-scripts\platforms\Windows\$FileName")
+  $MsbuildArgs += "-noLogo"
```
Comment (compnerd) [inline build.ps1:263]: is it `-noLogo`, `-NoLogo`, or `-nologo`? 🤔

File: build.ps1 [hunk 7]
```diff
@@ -898,6 +922,35 @@ function Build-SourceKitLSP($Arch)
     }
 }
 
+function Build-Installer()
+{
+  # Build-WiXProject toolchain.wixproj -Properties @{
+  #   DEVTOOLS_ROOT = "$ToolchainInstallRoot\";
+  #   TOOLCHAIN_ROOT = "$ToolchainInstallRoot\";
+  # }
```
Comment (compnerd) [inline build.ps1:930]: Is this meant to be disabled?

File: build.ps1 (hunk 6 above)
Comment (tristanlabelle) [reply build.ps1]: `noLogo` is ugly but the docs use that ¯\_(ツ)_/¯
```
C:\Program Files\Microsoft Visual Studio\2022\Community>msbuild --help
MSBuild version 17.5.0+6f08c67f3 for .NET Framework
Syntax:              MSBuild.exe [options] [project file | directory]
[...]
  -noLogo            Do not display the startup banner and copyright message.

File: build.ps1 (hunk 7 above)
Comment (tristanlabelle) [reply build.ps1:931]: Added:
```
# Currently fails due to _InternalSwiftScan paths

File: build.ps1 (hunk 6 above)
Comment (compnerd) [reply build.ps1]: We can follow the docs, but we don't need to like it!

**Review by compnerd — APPROVED** [review]
Nice, I still wish that this were three changes, but I'll live :-p.

**Pull request merged**